sources/web/datalab/jupyter.ts

```diff
@@ -137,7 +137,16 @@ function createJupyterServer(userId: string, resolved: (server: JupyterServer)=>
         server.proxy.on('proxyRes', responseHandler);
         server.proxy.on('error', errorHandler);
 
-        resolved(server);
+        tcp.waitUntilUsed(server.port).then(
```

Comment:
Maybe set a larger timeout value? Default is only 300ms and not sure Jupyter is quick enough to start up especially in a potentially heavy loaded VM.